iterators.rb

```diff
@@ -0,0 +1,132 @@
+# rubocop:disable  Metrics/ModuleLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    index = 0
+    while index < size
+
+      case self.class.name
+      when 'Hash' then yield(keys[index], self[keys[index]])
+      when 'Array' then yield(self[index])
+      when 'Range' then yield(to_a[index])
+      end
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    index = 0
+    while index < size
+      yield(self[index], index)
+      index += 1
+    end
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    array = []
+    my_each do |element|
+      array.push(element) if yield(element) == true
+    end
+    array
+  end
+  alias my_find_all my_select
+  alias my_filter my_select
+
+  def my_all?(pattern = nil)
+    is_equal = true
+    if pattern.nil?
+      if block_given?
+        my_each { |val| is_equal = false unless yield(val) or size.zero? }
+      else my_each { |val| is_equal = false unless val == true }
+      end
+    elsif pattern.is_a?(Module)
+      if pattern.is_a?(Regexp)
+        my_each { |val| is_equal = false unless pattern.match(val) }
+      else
+        my_each { |val| is_equal = false unless val.is_a?(pattern) }
+      end
+    end
+    is_equal
+  end
+
+  def my_any?
+    my_each do |val|
+      return true if yield(val)
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[nil, false, true, []].my_any? == [nil, false, true, []].any? => true
```
- Yours returns a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/80292761-32d3e800-8716-11ea-887c-8f8fa1b8a7fc.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689425-3f262300-6f67-11ea-88a9-2885ebab08d4.png)

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80292774-4f702000-8716-11ea-97e5-7abd7fc104aa.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689595-8f04ea00-6f67-11ea-9f51-4536c3f8049b.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/z/) == %w[dog door rod blade].any?(/z/) => false
```
- Yours retrieves an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80292786-6878d100-8716-11ea-99ae-6e058fc7c04d.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689742-cf646800-6f67-11ea-9e3e-eec7a960fe38.png)

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
[3,4,7,11].my_any?(3) == [3,4,7,11].any?(3) => true
```
- Yours retrieves an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80292803-85ad9f80-8716-11ea-8c66-846406acc434.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689883-10f51300-6f68-11ea-8a4c-319977bb3842.png)

Reply:
**Fixed**